src/Target/Php/Transpiler/Tag/TagTranspiler.php

```diff
@@ -39,11 +39,11 @@ public function transpile(TagNode $tagNode): string
     {
         $result = sprintf('<%s', $tagNode->tagName);
 
-        $attributeTranspiler = null;
+        $attributeTranspiler = new AttributeTranspiler(
+            scope: $this->scope
+        );
```

Comment:
Not sure about this one...

I take it, that your intention is to have a consistent way of instantiating subsequent transpilers (at least as far as the `TagTranspiler` is concerned). I would argue that this change introduces inconsistency rather than consistency, which runs counter this intention.

To unpack this, I'll take your commit message as a basis, which states:
> [...] we dont lazily instantiate objects in other places, so we might as well dont do it here ^^

Firstly, I'd argue that it's not entirely true that we do not lazily instantiate objects in other places, because in this very method - in a later set of statements - the `TagContentTranspiler` gets instantiated *conditionally*:
https://github.com/PackageFactory/PackageFactory.ComponentEngine/blob/bf0f15f7d52173ca609d44e68a6deb85630251f1/src/Target/Php/Transpiler/Tag/TagTranspiler.php#L52-L56

To have consistency, either both, the `AttributeTranspiler` and the `TagContentTranspiler` must be instantiated upfront, or both must be instantiated when needed (a.k.a. lazily). 

Secondly, if we'd go for upfront instantiation, I would furthermore argue that both transpiler instances should be bound to the `TagTranspiler` instance - thus be instantiated in `TagTranspiler`s constructor - rather than its `transpile`-method scope. 

Of course, this is not a must - but it would look like an obvious optimization step, which makes it likely that future refactoring will gravitate towards it. Over time this leads to a problematic consideration for constructor call-sites: What exactly is the cost of instantiation for a given transpiler class?

To simplify (or rather: avoid) this question, I apply a simple (though implicit) rule:
**Never instantiate a transpiler class, unless its `transpile`-method is guaranteed to be called**

Unfortunately, I see no way of making this rule explicit or enforce it structurally, but if followed strictly, it definitely escapes the erosive effect I've just described.

It is under this rule that I would consider lazy instantiation of the `AttributeTranspiler` more consistent in this case.

Reply:
Okay wow this is a well founded argumentation (and a nice read ^^)

I must admit, that I must have confused the instantiation of the transpilers with the resolvers (which happen to be eager instantiated at many places)

under this distinction (I must have mixed it up) transpiler !== typeresolver, I agree on your final statement and will revert this ;)

thanks for taking your time ^^